cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java

```diff
@@ -512,11 +513,13 @@ protected PersistentVolumeClaim createPersistentVolumeClaim(String name) {
                 .withRequests(requests)
                 .endResources()
                 .withStorageClassName(storage.getStorageClass())
-                .withNewSelector().withMatchLabels(storage.getSelector()).endSelector()
-                .endSpec()
-                .build();
+                .endSpec();
+
+        if (!storage.getSelector().isEmpty()) {
```

Comment:
I'm surprised `storage.getSelector()` is never null, actually (the tests pass without NPEs). I'd still be inclined to add a null check on `storage.getSelector()`.

Reply:
Good call! I think I've been writing too much Kotlin lately.